documentation/workflows.md

```diff
@@ -0,0 +1,118 @@
+# Working on the Project
+
+Better to be explicit/specific on topics than to wait for avoidable confusion due to ambiguity.
+
+## Principles
+
+1. Changes are NEVER done directly on `main`
+    - ALWAYS branch off of the target receiving branch (i.e. `main` for the majority of work)
+    - This allows for peer review and approval
+2. All quality gates SHOULD be passed before review
+    - Linting
+    - Type-checking
+    - Isolation testing: unit, integration, functional
+    - Testing coverage thresholds
+3. All changes must be reviewed and approved by another contributor
+    - An author can not "approve" their own changes
+    - Some areas might have [code owners](../.github/CODEOWNERS.md) assigned requiring specific people to review
+    - Contributors should aim for a ratio of 1/1 (PRs reviewed/opened) or better (greater than 1)
```

Comment:
I don't know if we should enforce any specificity around this as it is hard to enforce from an automation standpoint and doesn't really consider more ephemeral contributors from the community in the future.

Reply:
I will add more specificity. This _is_ only intended for internal contributors and is something I was passionate about in previous employment and saw good engagement in once made explicit.

Reply:
I'll remove actually. This is detail more for a team than a repo.